template/README.md

```diff
@@ -11,27 +11,97 @@ If you found this in `fredrikhgrelland/vagrant-hashistack`, you may be intereste
 
 Documentation on [parent repository](https://github.com/fredrikhgrelland/vagrant-hashistack#usage).
 
-## Customizing and using the vagrant box
+## Goals of this template
+This template aims to supply a developer with the tools necessary to do three things:
+- Build a docker image
+- Build a terraform module
+- Test and develop these two within a hashistack ecosystem
+
+To achieve this we utilise a vagrant-box built from [fredrikhgrelland/vagrant-hashistack](https://github.com/fredrikhgrelland/vagrant-hashistack/).
+
+## Where do I start?
+First of all I would recommend you read the section ["Using the template and box"](using-the-template-and-box) for a detailed look at the box's features and configurations. Below is a rough step-by-step going from a fresh template to a functioning terraform module. You can find example-code of everything below in [test_example/](test_example/).
+
+Step-by-step:
+1. Create a [Dockerfile](https://docs.docker.com/engine/reference/builder/) and put it in `docker/`.
+1. Use the box's startup playbooks to build and make the image available within the box. [Startup scheme](default-startup-scheme). [Example building image](test_example/dev/ansible/01_build_docker_image.yml).
+1. Create a [Nomad job specification](https://www.nomadproject.io/docs/job-specification). This should be placed in [conf/nomad/](conf/nomad/README.md). [Example hcl-file](test_example/conf/nomad/countdash.hcl)
+1. Use terraform to deploy this job to Nomad. Refer to ["Terraform module requirements"](terraform-module-requirements) on how to do this.
+
+## File structure
+- [conf/nomad/](conf/nomad/), nomad job specifications
+- [dev/](dev/), files used to control testing and development
+  - [dev/ansible](dev/ansible), playbooks run at startup
+  - [dev/vagrant/conf/](dev/vagrant/conf), files altering the vagrant-box  
+- [docker/](docker/), files used to build docker-image
+- [example/](example/), terraform-files using the terraform-module, [ref](terraform-module-requirements)
+- [test_example/](test_example/), a copy of template, filled with example-code
+
```

Comment:
```suggestion
```
After adding it I'm conflicted, I feel this is overkill. Comments before I remove it?

Reply:
```
|-- conf/						
|     |-- nomad/			# nomad job specifications
|   
|-- dev/				# files used to control testing and development
|     |-- ansible/			# playbooks run at startup
|	  |-- vagrant/			
|            |-- conf/		        # files altering the vagrant-box
|-- docker/				# files used to build docker-image
|
|-- example/				# terraform-files using the terraform-module
|
|-- test_example/			# a copy of template, filled with example-code

```

Reply:
Hm, this does not format nicely when I use it in the code, am I doing something wrong?

Reply:
Probably you need to adjust, my previous comments is a draft.

Reply:
Neha fixed this. https://github.com/fredrikhgrelland/vagrant-hashistack/pull/242#discussion_r473919647
Now fixed!